### PR TITLE
[Consensus 2.0] Additional metrics & logs

### PR DESCRIPTION
**Pull request by akichidis**
## Description 

More metrics and logs related to block receive/acceptance/commit , subscription etc

## Test plan 

CI

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: consensus/core/src/authority_service.rs [hunk 1]
```diff
@@ -406,6 +420,18 @@ impl<T: 'static + Clone + Send> Stream for BroadcastStream<T> {
...
+impl<T> Drop for BroadcastStream<T> {
+    fn drop(&mut self) {
+        let peer_hostname = &self.context.committee.authority(self.peer).hostname;
+        self.context
+            .metrics
+            .node_metrics
+            .subscriber_connections
+            .with_label_values(&[peer_hostname, "inbound"])
+            .set(0);
+    }
```
[the remark this answers is not in the document]
Comment (mwtian) [reply consensus/core/src/authority_service.rs:432]: Good idea. I think we should use separate metrics for inbound and outbound though so the metric as a whole makes sense.

File: consensus/core/src/subscriber.rs [hunk 2]
```diff
@@ -119,6 +112,13 @@ impl<C: NetworkClient, S: NetworkService> Subscriber<C, S> {
         let mut retries: i64 = 0;
         let mut delay = INITIAL_RETRY_INTERVAL;
         'subscription: loop {
+            context
```
Comment (mwtian) [inline consensus/core/src/subscriber.rs:115]: Good fix.

File: consensus/core/src/metrics.rs [hunk 3]
```diff
@@ -237,9 +267,15 @@ impl NodeMetrics {
                 "Designates whether the synchronizer scheduler task to fetch blocks is currently running",
                 registry,
             ).unwrap(),
-            fetched_blocks: register_int_counter_vec_with_registry!(
-                "fetched_blocks",
-                "Number of fetched blocks per peer authority via the synchronizer.",
+            synchronizer_fetched_blocks_by_peer: register_int_counter_vec_with_registry!(
+                "synchronizer_fetched_blocks_by_peer",
+                "Number of fetched blocks per peer authority via the synchronizer and also by block authority",
+                &["peer", "type"],
```
Comment (mwtian) [inline consensus/core/src/metrics.rs:275]: The cardinality of # peers * # authorities could be just too large (>10000). We can test in private testnet but for now I'm splitting them into two metrics.

**Review by mwtian — APPROVED** [review]

**Review by arun-koshy — APPROVED** [review]

File: consensus/core/src/block_manager.rs [hunk 4]
```diff
@@ -94,10 +99,17 @@ impl BlockManager {
...
 
         for block in blocks {
+            self.update_block_receive_metrics(&block);
+
             // Try to accept the input block.
+            let block_ref = block.reference();
             let block = match self.try_accept_one_block(block) {
                 TryAcceptResult::Accepted(block) => block,
                 TryAcceptResult::Suspended(ancestors_to_fetch) => {
+                    debug!(
```
[the remark this answers is not in the document]
Comment (akichidis) [reply consensus/core/src/block_manager.rs:110]: Agreed

File: consensus/core/src/authority_service.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (akichidis) [reply consensus/core/src/authority_service.rs:432]: Splitting them makes sense 👍

**Pull request merged**